yt/yt/server/node/exec_node/job_proxy_log_manager.cpp

```diff
@@ -0,0 +1,138 @@
+#include "job_proxy_log_manager.h"
+
+#include "bootstrap.h"
+#include "private.h"
+
+#include <yt/yt/server/lib/exec_node/config.h>
+#include <yt/yt/server/lib/scheduler/helpers.h>
+#include <yt/yt/server/node/cluster_node/config.h>
+
+#include <yt/yt/core/misc/fs.h>
+
+#include <util/datetime/base.h>
+#include <util/generic/ymath.h>
+#include <util/system/fstat.h>
+
+namespace NYT::NExecNode {
+
+static auto& Logger = ExecNodeLogger;
+
+////////////////////////////////////////////////////////////////////////////////
+
+TJobProxyLogManager::TJobProxyLogManager(IBootstrap* bootstrap)
+    : Bootstrap_(bootstrap)
+    , Config_(Bootstrap_->GetConfig()->ExecNode->JobProxyLogManager)
+    , Enabled_(Bootstrap_->GetConfig()->ExecNode->JobProxy->JobProxyLogging->Mode == EJobProxyLoggingMode::PerJobDirectory)
+    , Directory_(Config_->Directory)
+    , ShardingKeyLength_(Config_->ShardingKeyLength)
+    , LogsStoragePeriod_(Config_->LogsStoragePeriod)
+    , DirectoryTraversalConcurrency_(Config_->DirectoryTraversalConcurrency)
+    , AsyncSemaphore_(New<NConcurrency::TAsyncSemaphore>(Config_->DirectoryTraversalConcurrency.value_or(0)))
```

Comment:
Am I understand correctly that 0 in async semaphore means unlimited concurrency?

Reply:
I think TAsyncSemaphore doesn't support unlimited concurrency. Instead, 0 is used as a placeholder value here. And when the semaphore is acquired, there's the check that the DirectoryTraversalConcurrency value is not 0. If it's 0, semaphore is not acquired.